demo/lightning/image-segmentation/train.py

```diff
@@ -1,42 +1,223 @@
 import lightning as pl
 
-from model import Unet3DLightning
-from data import Unet3DDataModule
 from arguments import PARSER
+import time
+
+from typing import Any, Dict, Optional
+
+import torch
+import functools
+from dataflux_core import user_agent
+from google.cloud import storage
+from lightning.pytorch.plugins.io import CheckpointIO
 
 import ray.train.lightning
 from ray.train.torch import TorchTrainer
+from ray.train.lightning import RayFSDPStrategy
+from torch.distributed.fsdp.wrap import transformer_auto_wrap_policy
+from torch.distributed.fsdp import ShardingStrategy, BackwardPrefetch
+from transformers.models.gpt_neox.modeling_gpt_neox import GPTNeoXLayer
+from ray.train import RunConfig, CheckpointConfig
+from ray.train.lightning import RayLightningEnvironment, RayTrainReportCallback, prepare_trainer
+import os
+import time
+from typing import Tuple
+
+import torch
+from lightning import Trainer
+from lightning.pytorch.callbacks import ModelCheckpoint
+from torch import Tensor
+import pandas as pd
+from datasets import load_dataset
+from transformers import AutoTokenizer, AutoModelForCausalLM
+
+class DollyV2Model(pl.LightningModule):
+    def __init__(self, lr=2e-5, eps=1e-8):
+        super().__init__()
+        self.save_hyperparameters()
+        self.lr = lr
+        self.eps = eps
+        self.model = AutoModelForCausalLM.from_pretrained("databricks/dolly-v2-7b")
+
+    def forward(self, batch):
+        outputs = self.model(
+            batch["input_ids"],
+            attention_mask=batch["attention_mask"],
+            labels=batch["labels"]
+        )
+        return outputs.loss
+
+    def training_step(self, batch, batch_idx):
+        loss = self.forward(batch)
+        self.log("train_loss", loss, prog_bar=True, on_step=True)
+        return loss
+
+    def configure_optimizers(self):
+        if self.global_rank == 0:
+            print(self.trainer.model)
```

Comment:
nit: self.log() to align with the logging method in this class. Additionally, you can potentially use [rank_zero_only](https://lightning.ai/docs/pytorch/stable/extensions/logging.html#automatic-logging) to save the "if".